source/grid/grid_tools.cc

```diff
@@ -3012,6 +3012,7 @@ namespace GridTools
           }
         else
           {
+#if !defined(__INTEL_LLVM_COMPILER) || BOOST_VERSION >= 108100
```

Comment:
At the very least this would require a comment that explains why the code below can actually be omitted. I have to admit that I have no idea what the code does, but it is not obvious why we can skip certain code blocks if we use a specific compiler :-)